### PR TITLE
core(driver): remove unused goOffline/goOnline methods

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
Removes some unused parts of driver I noticed while refactoring pieces for #11313. I don't recall if we ever established a semver policy for removing esoteric driver methods, but if there was a conversation I missed about this please feel free to correct me :) 

Given how much of a delineation we made for non-public artifacts that are technically still accessible, this seems perfectly in line with that policy.

**Related Issues/PRs**
ref #11313

### CODE REVIEW

**Comment by brendankenny** [comment]
#578 📆 ⏳ 😢 

Maaaaaybe it's worth keeping? It does seem like the kind of thing custom gatherers might use...but I'm also 100% for deleting unused things, and that's winning out for me right now :) Also I've never loved that call hierarchy.

File: lighthouse-core/lib/emulation.js [hunk 1]
```diff
@@ -7,14 +7,6 @@
 
 /** @typedef {import('../gather/driver.js')} Driver */
 
-const OFFLINE_METRICS = {
-  offline: true,
-  // values of 0 remove any active throttling. crbug.com/456324#c9
```
Comment (brendankenny) [inline lighthouse-core/lib/emulation.js:12]: this seems like the only tricky knowledge that could be lost if trying to recreate going offline, but based on that comment is it still true/useful today?

Comment (patrickhulce) [reply lighthouse-core/lib/emulation.js:12]: Based on the [immediate followup commit](https://codereview.chromium.org/1039213002/patch/1/10001), it looks like future attempts should be fine with just `offline: true`.

If we ever need to find this again, we know the file to use git history on :)

**Review by adamraine — APPROVED** [review]
Looks good, but I don't have any strong opinions on keeping some of these.

**Comment by patrickhulce** [comment]
Let the record show that @paulirish +1'd this in person but did not use the 👍 reaction on any comment or PR description as of  2/25/2021 6:05pm CST despite claims to the contrary.


:)

**Comment by paulirish** [comment]
> Let the record show that @paulirish +1'd this in person but did not use the 👍 reaction on any comment or PR description as of 2/25/2021 6:05pm CST despite claims to the contrary.
> 
> :)

i don't know what you're talkin about…

image

;)

**Pull request merged**